### PR TITLE
Sync ``libasr`` with LFortran

### PR DESCRIPTION
**Pull request by czgdp1807**
Corresponding LFortran PR - https://github.com/lfortran/lfortran/pull/1325

### CODE REVIEW

File: src/libasr/asdl_cpp.py [hunk 1]
```diff
@@ -2529,6 +2559,9 @@ def main(argv):
         subs["MOD"] = "LPython::AST"
         subs["mod"] = "ast"
         subs["lcompiler"] = "lpython"
+    elif subs["MOD"] == "AST":
```
Comment (czgdp1807) [inline src/libasr/asdl_cpp.py:2562]: Shouldn't it be `"LFORTRAN"` instead of `"AST"`?

Comment (certik) [reply src/libasr/asdl_cpp.py:2562]: We haven't been consistent in the asdl if I remember well, so I would not do any big changes in the sync PRs, but we can fix things up later.

File: src/libasr/codegen/wasm_to_x64.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@ pushed back onto the stack.
 
 One of the reasons to use stack to pass function arguments is that,
 it allows us to define and call functions with any number of parameters.
-As registers are limited in number, if we use them to pass function arguments,
+As registers are limited in number, if we use them to pass function arugments,
```
Comment (certik) [inline src/libasr/codegen/wasm_to_x64.cpp:24]: I think some of the WASM stuff should be reverted, I think @Shaikh-Ubaid mostly worked in LPython.

**Comment by certik** [comment]
Otherwise it looks good.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 3]
```diff
@@ -4651,58 +4800,14 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         llvm::Value *left_val = tmp;
         this->visit_expr_wrapper(x.m_right, true);
         llvm::Value *right_val = tmp;
-        llvm::Value *zero, *cond;
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_llvm.cpp:4654]: #1506 to be reverted.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 4]
```diff
@@ -5221,16 +5331,8 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_Assert(const ASR::Assert_t &x) {
-        if (compiler_options.emit_debug_info) debug_emit_loc(x);
         this->visit_expr_wrapper(x.m_test, true);
         create_if_else(tmp, []() {}, [=]() {
-            if (compiler_options.emit_debug_info) {
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_llvm.cpp:5227]: #1344 this is to be reverted.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 5]
```diff
@@ -2889,27 +2992,20 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                             ptr->setAlignment(align);
                         }
                     }
+
                     llvm_symtab[h] = ptr;
                     fill_array_details_(ptr, m_dims, n_dims,
                         is_malloc_array_type,
                         is_array_type, is_list, v->m_type);
-                    ASR::expr_t* init_expr = v->m_symbolic_value;
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/codegen/asr_to_llvm.cpp:2896]: #1445 this is to be reverted

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 6]
```diff
@@ -1255,304 +1202,3 @@ LFORTRAN_API int32_t _lpython_get_argc() {
 LFORTRAN_API char *_lpython_get_argv(int32_t index) {
     return argv[index];
 }
-
-// << Command line arguments << ------------------------------------------------
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/runtime/lfortran_intrinsics.c:1259]: #1344

File: src/libasr/pass/print_list.cpp [hunk 7]
```diff
@@ -68,275 +52,148 @@ class PrintListVisitor
...
 
    public:
-   Vec<ASR::stmt_t*> print_pass_result_tmp;
     PrintListVisitor(Allocator &al, const std::string &rl_path_)
         : PassVisitor(al, nullptr), rl_path(rl_path_) {
         pass_result.reserve(al, 1);
-        print_pass_result_tmp.reserve(al, 1);
     }
 
-    void print_list_helper(ASR::expr_t *list_expr, ASR::expr_t *sep_expr,
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/print_list.cpp:78]: #1417

File: src/libasr/compiler_tester/tester.py [hunk 8]
```diff
@@ -358,12 +354,8 @@ def tester_main(compiler, single_test):
                         help="Exclude specific backends, only works when -b is not specified"),
     parser.add_argument("--no-llvm", action="store_true",
                         help="Skip LLVM tests")
-    parser.add_argument("--skip-run-with-dbg", action="store_true",
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/compiler_tester/tester.py:361]: #1463

**Review by Pranavchiku — APPROVED** [review]
Looks good to me!

File: src/libasr/asr_utils.h [hunk 9]
```diff
@@ -2100,9 +2443,18 @@ class ReplaceReturnWithGotoVisitor: public ASR::BaseStmtReplacer<ReplaceReturnWi
...
+            return true;
+        }
+    }
+    return false;
+}
+
 // Singleton LabelGenerator so that it generates
 // unique labels for different statements, from
-// wherever it is called (be it ASR passes, be it
+// whereever it is called (be it ASR passes, be it
```
Comment (Shaikh-Ubaid) [inline src/libasr/asr_utils.h:2457]: I think the previous spelling was correct. (https://prowritingaid.com/grammar/1000219/Whereever-vs-wherever%E2%80%94what-is-the-difference)

**Comment by Shaikh-Ubaid** [comment]
For the files `asr_to_wasm.cpp`, `asr_to_x86.cpp`, `wasm_assembler.h`, `wasm_decoder.h`, `wasm_to_wat.cpp`, `wasm_to_x64.cpp`, `wasm_to_x86.cpp`, `wasm_utils.h`, `x86_assembler.cpp`, `x86_assembler.h`, please, could you keep the same version as in current `lpython` master? And then we have to add the changes in https://github.com/lfortran/lfortran/pull/1216 and https://github.com/lfortran/lfortran/pull/1304.

**Review by Shaikh-Ubaid — APPROVED** [review]
It looks good to me. Thank you so much for this!

**Review by Thirumalai-Shaktivel — COMMENTED** [review]
LGTM, Thank you! 😃

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Review by Smit-create — APPROVED** [review]
Thanks, @czgdp1807. LGTM!!

**Pull request merged**